### PR TITLE
Compiler html like tags

### PR DESCRIPTION
**Pull request by csansoon**
🚂 TRENECITO
Destination: #8 

---
Parse any HTML-like tags. Tags contain a name, and an array of attributes:
```html
<name attr1=value1 attr2=value2> content </name>
```
Attributes can have the following values:
 - If a value is provided and wrapped in quotation marks, it is returned as a `Text` node.
 - If a value is provided and wrapped in custom logic brackets (`{{ }}`), it is returned as a `MustacheTag` with the correspondent expression.
 - If a value is provided as a single word with no wrap, it is returned as a `Text` node.
 - If no value is provided (and there's no = sign either), it is returned as a `true` value.

There is no logic in the Parser about what types of tags are there or what attributes do they expect. That's the compiler's responsibility.

---
Last stop: #8
Next station: #10

### CODE REVIEW

**Comment by geclos** [comment]
> If a value is provided as a single word with no wrap, it is returned as a Text node.

I wouldn't do this? I know it's cool but it increases the api surface for no added benefit (not having to write quotes is not a benefit) and it introduces some confusion `name=true` could be expected to parse true as a boolean instead of a string whereas name='true' is crystal clear.

File: packages/compiler/src/parser/parser.test.ts [hunk 1]
```diff
@@ -274,4 +275,138 @@ world */
       ` hello\n  ${CUSTOM_TAG_START}#if condition${CUSTOM_TAG_END}then${CUSTOM_TAG_START}/if${CUSTOM_TAG_END}\nworld `,
     )
   })
+
+  it('Allows tag comments', () => {
+    const fragment = parse('<!-- hello -->')
```
Comment (geclos) [inline packages/compiler/src/parser/parser.test.ts:280]: very cool!

File: packages/compiler/src/parser/state/tag.ts [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+				type: 'Text',
+				raw: '',
+				data: ''
+			};
+		} else {
+			currentChunk.raw += parser.template[parser.index++];
+		}
+	}
+	parser.error(PARSER_ERRORS.unexpectedEof);
+}
```
Comment (geclos) [inline packages/compiler/src/parser/state/tag.ts:219]: is this code copied from somewhere?

Comment (andresgutgon) [reply packages/compiler/src/parser/state/tag.ts:219]: don't worry when he rebase with main prettier will complain

**Comment by csansoon** [comment]
> > If a value is provided as a single word with no wrap, it is returned as a Text node.
> 
> I wouldn't do this? I know it's cool but it increases the api surface for no added benefit (not having to write quotes is not a benefit) and it introduces some confusion `name=true` could be expected to parse true as a boolean instead of a string whereas name='true' is crystal clear.

The correct approach to setting it as `true` would be with `name={{true}}`, as it is a logical value.

**Review by andresgutgon — DISMISSED** [review]

**Comment by geclos** [comment]
> The correct approach to setting it as true would be with name={{true}}, as it is a logical value.

I understand but the point is that it's confusing and there's no point on having this feature since you can already define both booleans and strings with  `foo={{}}` syntax

**Pull request merged**